File: Placement Sheet/LinkedList/BS/LinkedListStack.java
public class LinkedListStack {
    static class Node {
        private int data;
        private Node next;

        Node(int data) {
            this.data = data;
            this.next = null;
        }

        int getData() {
            return data;
        }

        Node getNext() {
            return next;
        }

        void setNext(Node node) {
            this.next = node;
        }
    }

    private Node top = null;
    private Node start = null;

    // Push...
    public void push(int data) {
        Node newNode = new Node(data);
        if (top == null) {
            start = newNode;
        } else {
            top.setNext(newNode);
        }
        top = newNode;
    }

    // Peek...
    public void peek() {
        if (top == null) {
            System.out.println("Stack is empty...");
            return;
        }
        System.out.println("Top element: " + top.getData());
    }

    // Pop...
    public void pop() {
        if (top == null) {
            System.out.println("Stack is empty...");
            return;
        }
        int value = top.getData();

        // Single element case
        if (start == top) {
            start = null;
            top = null;
        } else {
            Node currNode = start;
            while (currNode.getNext() != top) {
                currNode = currNode.getNext();
            }
            currNode.setNext(null);
            top = currNode;
        }

        System.out.println("Popped element: " + value);
    }

    // Display...
    public void display() {
        if (top == null) {
            System.out.println("Stack is empty...");
            return;
        }
        Node currNode = start;
        java.util.Stack<Integer> stack = new java.util.Stack<>();
        while (currNode != null) {
            stack.push(currNode.getData());
            currNode = currNode.getNext();
        }

        while (!stack.isEmpty()) {
            System.out.println("|" + stack.pop() + "|");
        }
    }

    public static void main(String[] args) {
        LinkedListStack stack = new LinkedListStack();
        stack.push(1);
        stack.push(2);
        stack.push(3);
        stack.push(4);
        stack.push(5);
        stack.push(6);
        stack.peek();
        stack.pop();
        stack.peek();
        stack.display();
    }
}

File: Placement Sheet/LinkedList/BS/LinkedListStack.ts
class NodeLL {
    private data: number;
    private next: NodeLL | null;

    constructor(data: number) {
        this.data = data;
        this.next = null;
    }

    getData(): number {
        return this.data;
    }

    getNext(): NodeLL | null {
        return this.next;
    }

    setNext(node: NodeLL | null): void {
        this.next = node;
    }
}

class LinkedlistStack {
    private top: NodeLL | null = null;
    private start: NodeLL | null = null;

    //Push...
    push(data: number): void {
        let newNode: NodeLL | null = new NodeLL(data);
        if (this.top == null) {
            this.start = newNode;
        }
        this.top?.setNext(newNode);
        this.top = newNode;
    }

    //Peek...
    peek(): void {
        if (this.top == null) {
            console.log("Stack is empty...");
            return;
        }
        const result: number = this.top.getData();
        console.log(`Top element: ${result}`);
    }

    //Pop...
    pop(): void {
        if (this.top == null) {
            console.log("Stack is empty...");
            return;
        }
        const value: number = this.top.getData();
        let currNode: NodeLL | null = this.start;
        while (currNode?.getNext() !== this.top) {
            currNode = currNode?.getNext()!;
        }
        currNode.setNext(this.top.getNext());
        this.top = currNode;
        console.log(`Poped element: ${value}`);
    }

    //Display...
    display(): void {
        if (this.top == null) {
            console.log("Stack is empty...");
            return;
        }
        let currNode: NodeLL | null = this.start;
        let result: number[] = [];
        while (currNode) {
            result.push(currNode.getData());
            currNode = currNode.getNext();
        }
        let length: number = result.length - 1;
        while (length >= 0) {
            console.log("|" + result[length] + "|");
            length--;
        }
    }
}

let Stack: LinkedlistStack = new LinkedlistStack();
Stack.push(1);
Stack.push(2);
Stack.push(3);
Stack.push(4);
Stack.push(5);
Stack.push(6);
Stack.peek();
Stack.pop();
Stack.peek();
Stack.display();

export { };